src/content/react-hooks-color-picker.md

```diff
@@ -0,0 +1,187 @@
+
+This will be a very simple demonstration. At the top, we'll have a `ColorBox` component which will show the current color and change dynamically with user interaction.
+Below that, there will be three range sliders to adjust the `hue`, `saturation` and `brightness`.
+
+## Adding state
+
+As described above, we can use hooks to control the state of each range picker dimension. First, let's define our state in the App body, like so:
+
+```js
+const [hue, setHue] = useState(210);
+const [saturation, setSaturation] = useState(50);
+const [lightness, setLightness] = useState(50);
+```
+
+We can then create a const called `currentColor`, which we can pass later as a prop to our `ColorBox` component:
+
+```js
+const currentColor = `hsl(${hue}, ${saturation}%, ${lightness}%)`;
+```
+
+## RangeInput Component
+
+Let's first create a range input component, since it will be reused 3 times.
+
+```js
+const RangeInput = ({ label, value, max = 100, onChange }) => {
+  return (
+    <div>
+      {label}{' '}
+      <input
+        type="range"
+        value={value}
+        min={0}
+        max={max}
+        onChange={e => {
+          onChange(e.target.value);
+        }}
+      />
+    </div>
+  );
+};
+```
+
+I set `max = 100` by default, since for two of the range sliders, 100 is the max. For hue, the max value will go to 360, so we can pass props as such to define this one:
+
+```js
+<RangeInput label="Hue" value={hue} onChange={setHue} max={360} />
+```
+
+The other inputs will look like this:
+
+```js
+<RangeInput
+  label="Saturation"
+  value={saturation}
+  onChange={setSaturation}
+/>
+<RangeInput
+  label="Brightness"
+  value={lightness}
+  onChange={setLightness}
+/>
+```
+
+## ColorBox component
+
+For the `ColorBox` component, I will use `styled-components`. You can add this dependency by doing `npm install styled-components`. With this library, we can not only create the basic styled, but pass the current color we defined above as props, as such:
+
+```js
+const ColorBox = styled.div`
+  background-color: ${({ currentColor }) => currentColor};
+  margin: 0 auto 16px;
+  height: 50px;
+  width: 50px;
+`;
+```
+
+For more information about how to use styled-components, see <a href="https://www.styled-components.com/docs/basics#adapting-based-on-props" target="_blank" rel="noopener">Adapting styled components based on props</a>
+
+We can then, pass the current color through props, and the background color will automatically update when the value of hue, saturation or brightness changes.
+
+```js
+<ColorBox currentColor={currentColor} />
+```
+
+## Summing up
+
+And there you have it! With a few lines of code you have your own custom color picker made with react.
+
+Of course, `useState` is not the only React Hook out there. To go further with your React Hooks studies, please check out the official <a href="https://reactjs.org/docs/hooks-intro.html" target="_blank" rel="noopener noreferrer">React Hooks docs</a>at reactjs.org.
```

Comment:
@eliotc1986 

Why you use not Markdown writing rule but html a tag?
If you want to add attribute "target" and "rel", I think it's better to use "gatsby-remark-external-links".
Take a look at following site: https://pastora.jp/posts/set-target-blank-to-the-external-link/

Reply:
@yosi-q Thank you for the comment!
For this, I don't want to add another dependency. Plus, Markdown is fully capable of handling html5 syntax. https://www.markdownguide.org/basic-syntax/

So, is it okay if I skip this?